model/lifecycle.go

```diff
@@ -1294,9 +1263,26 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 		CommitQueueMerge:    buildVarTask.CommitQueueMerge,
 		IsGithubCheck:       isGithubCheck,
 		DisplayTaskId:       utility.ToStringPtr(""), // this will be overridden if the task is an execution task
-		// TODO (EVG-16169): set ExecutionPlatform based on whether or not the task
-		// is running in a host or container.
-		ExecutionPlatform: task.ExecutionPlatformHost,
+	}
+	// is running in a host or container.
+	t.ExecutionPlatform = shouldRunOnContainer(buildVarTask.RunOn, buildVariant.RunOn, project.Containers)
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting service flags")
+	}
+	if t.ExecutionPlatform == task.ExecutionPlatformContainer {
+		if flags.ContainerConfigurationsDisabled {
+			return nil, errors.Errorf("container configurations are disabled; task '%s' cannot run", t.DisplayName)
+		} else {
```

Comment:
It's more idiomatic in Go to omit the `else` when the `if` block already returns.